zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneService.java

```diff
@@ -88,13 +165,305 @@ private ConnectionDetails buildConnectionDetails(String zoneName, IdentityZone i
+        String zoneSigningKey = "";
+        try {
+            zoneSigningKey = createSigningKey();
+        } catch (Exception e) {
+            logger.error("Unable to create signingKey for subdomain: " + subdomain + " Exception is : " + e.getMessage());
+            throw new OrchestratorZoneServiceException("Unexpected exception while create signingKey for subdomain: " + subdomain);
+        }
+        return zoneSigningKey;
+    }
+
+    private String getCustomSubdomain(final String subdomain) throws OrchestratorZoneServiceException {
+        if (subdomain == null) {
+            return null;
+        }
+        String subDomain = subdomain;
+        if (!StringUtils.hasText(subDomain)) {
+            throw new OrchestratorZoneServiceException(
+                "The \"" + UAA_CUSTOM_SUBDOMAIN + "\" field cannot contain spaces or cannot be blank.");
+        }
+        if (!SUBDOMAIN_PATTERN.matcher(subDomain).matches()) {
+            throw new OrchestratorZoneServiceException("The \"" + UAA_CUSTOM_SUBDOMAIN
+                                                       + "\" is invalid. Special characters are not allowed in the subdomain name except hyphen which can be specified in the middle.");
+        }
+        return subDomain;
+    }
+
+    private void createClient(final String id, final String authorities, final String clientId,
+                              final String clientSecret,
+                              final String grantTypes, final String resourceIds, final String scopes) {
+        BaseClientDetails clientDetails = new BaseClientDetails(clientId, resourceIds, scopes, grantTypes, authorities);
+        clientDetails.setClientSecret(clientSecret);
+        ClientDetails details = clientDetailsValidator.validate(clientDetails, Mode.CREATE);
+        clientDetailsService.create(details, id);
+    }
+
+    public void createUserGroups(IdentityZone zone) {
+        UserConfig userConfig = zone.getConfig().getUserConfig();
+        if (userConfig != null) {
+            List<String> defaultGroups = ofNullable(userConfig.getDefaultGroups()).orElse(Collections.emptyList());
+            for (String group : defaultGroups) {
+                groupProvisioning.createOrGet(
+                    new ScimGroup(
+                        null,
+                        group,
+                        zone.getId()
+                    ),
+                    zone.getId()
+                                             );
+            }
+        }
+    }
+
+    private List<String>  createDeploymentSpecificLogoutWhiteList()
+    {
+        String runDomainFQDN = getRunDomainFromUAADomain();
+        return (StringUtils.isEmpty(runDomainFQDN))  ? Collections.singletonList("http*://**") :
+               Collections.singletonList("http*://**" + runDomainFQDN);
+    }
+
+    /**
+     * Remove all characters till first dot
+     * @return
+     */
+    private String getRunDomainFromUAADomain() {
```

Comment:
There is no "run" domain in EKS. What's the equivalent in EKS? Need look into how logout whitelist works and what it should be for EKS.

Reply:
As discussed, we have a story for UAA logout url whitelist using self-url pattern extraction.